### PR TITLE
Cones supported by solvers

### PR DESCRIPTION
**Pull request by tcunis**
This PR changes the abbreviations of the different cones as well as provides an interface to query the cone types supported by a specific solver.

The following cones are supported (see also #21):

1. Convex cones of scalars, vectors, and vectorized matrices in `casos.conic` and `casos.sdpsol`:
  - `lin` (previously `l`): element-wise inequalities
  - `lor` (previously `q`): Lorentz (quadratic, second-order) cone
  - `rot` (previously `r`): Rotated Lorentz cone (not yet implemented for `SCS`)
  - `psd` (previously `s`): Cone of positive semi-definite matrices

2. Polynomial cones in `casos.sossol` and `casos.qcsossol`:
  - `lin` (previously `l`): coefficient-wise inequalities (reduces to element-wise inequalities for entries of degree zero)
  - `sos` (previously `s`): cone of sum-of-squares polynomials (relax to PSD)

On a related note, the demo has been reorganized.

### CODE REVIEW

File: examples/roaest_gtm.m
Comment (Fabian-Geyer) [inline examples/roaest_gtm.m:1]: rename 's' to 'sos' required

**Review by Fabian-Geyer — CHANGES_REQUESTED** [review]
roaest_gtm.m still in legacy mode

**Comment by renatoloureiro** [comment]
To run and test all demos in the repository and additional examples ( SDPs ) 
[test_file_review.zip](https://github.com/iFR-ACSO/casos/files/14787896/test_file_review.zip)

File: examples/roaest_gtm.m
Comment (tcunis) [reply examples/roaest_gtm.m:1]: thanks for noticing!

**Review by Fabian-Geyer — APPROVED** [review]

**Pull request merged**